api/app/files/api.py

```diff
@@ -45,9 +46,33 @@ def get(self):
             return send_file(temp.name, as_attachment=True, attachment_filename=args['filename'], mimetype='application/pdf')
 
 
+
+class FileAPIImproved(FileUploadMixin, restful.Resource):
+    def get(self, file_id):
+        # TO DO: Query the File table
+
+        req_parser = reqparse.RequestParser()
+        req_parser.add_argument('filename', type=str, required=True)
+        args = req_parser.parse_args()
+        LOGGER.debug("FileUpload GET args: {}".format(args))
+
+        bucket = _get_storage_bucket()
+
+        blob = bucket.blob(args['filename'])
+
+        # TO DO : Rename the file and set the appropriate mime_type
+
+        with tempfile.NamedTemporaryFile() as temp:
+            blob.download_to_filename(temp.name)
+            return send_file(temp.name, as_attachment=True, attachment_filename=args['filename'], mimetype='application/pdf')

```

Comment:
Then instead of `args`, we will use the `file` object we just retrieved from the database to set the `attachment_filename` and `mimetype`.